coffee/visitors/inspectors.py

```diff
@@ -504,15 +531,15 @@ class FindInstances(Visitor):
     Visit the tree and return a dict mapping types to a list of
     instances of that type in the tree.
 
-    :arg types: list of types or single type to search for in the
-          tree.
-    :arg stop_when_found: optional, don't traverse the children of
-          matching types.
+    :arg types: list of types or single type to search for in the tree.
+    :arg stop_when_found: optional, don't traverse the children of matching types.
+    :arg with_parent: optional, track also the parent of the matching type.
     """
 
-    def __init__(self, types, stop_when_found=False):
+    def __init__(self, types, stop_when_found=False, with_parent=False):
```

Comment:
Wouldn't it be simpler to just collect the parent information, and then drop it as a post-processing step on the resulting dictionary?